src/canmatrix/canmatrix.py

```diff
@@ -584,13 +585,92 @@ def __attrs_post_init__(self):
         if self.id != self.id & mask:
             raise ArbitrationIdOutOfRange('ID out of range')
 
+    @property
+    def j1939_pgn(self):
+        return self.pgn
+
+    @property
+    def pgn(self):
+        if not self.extended:
+            raise J1939needsExtendedIdetifier
+
+        ps = (self.id >> 8) & 0xFF
+        pf = (self.id >> 16) & 0xFF
+        _pgn = pf << 8
+        if pf >= 240:
+            _pgn += ps
+        return _pgn
+
+    @property
+    def j1939_tuples(self):  # type: () -> typing.Tuple[int, int, int]
```

Comment:
Consider to rename to `j1939_tuple` as it is single tuple carrying three values.